### DISCUSSION TITLE
Way to pass extra args to docker container image extension?

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#25467] [Q&A] [answered]
I'm looking to build multi-arch images and am currently using the Quarkus container-image-docker to build the image. I'm pushing the image separately.

Is there a way to pass extra flags to the docker command it uses under the covers? Specifically, the `--platform` flag.

**Comment by edeandrea** [comment, marked as answer]
Anyone?

/cc: @geoand (because I wasn't sure who else to cc & it doesn't seem like the Quarkus bot did anything on this one).

**Reply by geoand** [reply]
We don't have that, but it should be easy to add it. We would need to add the field in `io.quarkus.container.image.docker.deployment.DockerConfig` and make use of it in `io.quarkus.container.image.docker.deployment.DockerProcessor`

**Reply by edeandrea** [reply]
Do you see anything wrong with me adding this to `DockerConfig`? Basically just allow what the config says to "pass through" to the executable. Or would you see that as a potential security hole (i.e. injection)?

```java
/**
  * Additional command line arguments to be passed to the {@link #executableName}.
 */
@ConfigItem
public Optional<List<String>> additionalCommandArgs;
```

I thought about just adding a `platform` config item specifically, but that option *ONLY* exists if you're using `docker buildx build`.

**Reply by edeandrea** [reply]
@geoand I submitted #25542 for this. We can discuss the approach there.
